### PR TITLE
47691 SiS Enabled Sign-in Redirect Bug Fix

### LINKED ISSUE #47691: Protected Route Sign-In redirect update
## Issue Description
When a unauthenticated(not signed in yet) user attempts to access a protected route they are redirected to the home page with SSOe sign in. This needs to be updated to redirect to SiS sign in page. 

## Tasks
- [ ] Update the protected route to redirect to SiS component.

## Acceptance Criteria
- [ ] When an unauthenticated user attempts to access a protected route they are redirected to the SiS sign in page.

### PR DESCRIPTION
**Pull request by acaruso-oddball**
## Description
Currently when an unauthenticated user tries to access a route that requires authentication, they are redirected to a version of the Login Modal where SiS is not enabled. This PR updates the component that does such redirection (the RequiredLogInView component) to allow for the use of SiS when users are routed to the LoginModal. 

## Original issue(s)
closes [Protected Route sign-in redirect](https://github.com/department-of-veterans-affairs/va.gov-team/issues/47691)

## Testing done
Unit 

## Acceptance criteria
- [x] When an unauthenticated user attempts to access a page that requires authentication they are redirected to a SiS enabled SignIn Page. 

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by it-harrison — APPROVED** [review]

File: src/platform/user/tests/authorization/components/RequiredLoginView.unit.spec.jsx
Comment (AGarcia-Clarity) [inline src/platform/user/tests/authorization/components/RequiredLoginView.unit.spec.jsx:149]: Clean up uncommented code.

File: src/platform/user/tests/authorization/components/RequiredLoginView.unit.spec.jsx
Comment (AGarcia-Clarity) [inline src/platform/user/tests/authorization/components/RequiredLoginView.unit.spec.jsx:9]: Clean up uncommented code

Comment (acaruso-oddball) [reply src/platform/user/tests/authorization/components/RequiredLoginView.unit.spec.jsx]: thx

File: src/platform/user/tests/authorization/components/RequiredLoginView.unit.spec.jsx
Comment (acaruso-oddball) [reply src/platform/user/tests/authorization/components/RequiredLoginView.unit.spec.jsx]: thx

File: src/applications/post-911-gib-status/containers/Post911GIBStatusApp.jsx
Comment (acaruso-oddball) [reply src/applications/post-911-gib-status/containers/Post911GIBStatusApp.jsx:1]: This file will need to be updated by the appropriate team to allow for use of the VA telephone component.

**Comment by AGarcia-Clarity** [comment]
@acaruso-oddball Pull in latest as I believe the `health-care-questionnaire` was removed and I can finish reviewing.

**Review by Mottie — APPROVED** [review]

File: src/platform/user/authorization/components/RequiredLoginView.jsx
Comment (adamwhitlock1) [inline src/platform/user/authorization/components/RequiredLoginView.jsx:175]: Out of curiosity, if there a reason you prefer using `connect` over the Redux hooks like useSelector?

**Review by adamwhitlock1 — APPROVED** [review]

**Review by acrollet — APPROVED** [review]
review should no longer be required from patient-check-in now that health-care-questionnaire has been removed, leaving this approval so PR won't be blocked.

**Review by asg5704 — APPROVED** [review]

**Review by vbahinwillit — COMMENTED** [review]
VAOS changes LGTM

**Comment by acaruso-oddball** [comment]
Hi folks, can I get a review from the following teams, @department-of-veterans-affairs/vsa-debt-frontend @department-of-veterans-affairs/vfs-mhv-secure-messaging @department-of-veterans-affairs/govcio-vfep-codereviewers

**Review by powellkerry — APPROVED** [review]

**Review by mmoyer-va — APPROVED** [review]

**Review by it-harrison — APPROVED** [review]

**Review by simiadebowale — APPROVED** [review]
LGTM

**Comment by acaruso-oddball** [comment]
Hello, can I get a review from @department-of-veterans-affairs/vsa-debt-frontend and @department-of-veterans-affairs/govcio-vfep-codereviewers thx

**Comment by AGarcia-Clarity** [comment]
VSA Debt: @ScottyRJames @Dr-Pongo 
GOVCIO VFEP: @markkvba @jsimonVA 
@ryanshaw 

Can you all please review this thank you 👍

**Review by markkvba — APPROVED** [review]

**Review by Dr-Pongo — COMMENTED** [review]
Tagging @department-of-veterans-affairs/benefits-team-1-frontend as well to notify them of claims updates! I'm going to look into updating codeowners to help clear things up as well

**Pull request merged**
